miner/worker.go

```diff
@@ -1451,22 +1495,74 @@ func (w *worker) fillTransactions(interrupt *atomic.Int32, env *environment) err
 		}
 	}
 
-	// Fill the block with all available pending transactions.
-	if len(prioPlainTxs) > 0 || len(prioBlobTxs) > 0 {
-		plainTxs := newTransactionsByPriceAndNonce(env.signer, prioPlainTxs, env.header.BaseFee, &w.interruptBlockBuilding)
-		blobTxs := newTransactionsByPriceAndNonce(env.signer, prioBlobTxs, env.header.BaseFee, &w.interruptBlockBuilding)
+	var plainTxsPrio, blobTxsPrio *transactionsByPriceAndNonce
+	var plainTxsNormal, blobTxsNormal *transactionsByPriceAndNonce
+	{
+		if len(prioPlainTxs) > 0 || len(prioBlobTxs) > 0 {
+			plainTxsPrio = newTransactionsByPriceAndNonce(env.signer, prioPlainTxs, env.header.BaseFee, &w.interruptBlockBuilding)
+			blobTxsPrio = newTransactionsByPriceAndNonce(env.signer, prioBlobTxs, env.header.BaseFee, &w.interruptBlockBuilding)
+		}
+		if len(normalPlainTxs) > 0 || len(normalBlobTxs) > 0 {
+			heapInitTime := time.Now()
+			plainTxsNormal = newTransactionsByPriceAndNonce(env.signer, normalPlainTxs, env.header.BaseFee, &w.interruptBlockBuilding)
+			blobTxsNormal = newTransactionsByPriceAndNonce(env.signer, normalBlobTxs, env.header.BaseFee, &w.interruptBlockBuilding)
+			txHeapInitTimer.Update(time.Since(heapInitTime))
+		}
+		// Flatten pending txs into a slice (no capping)
+		txs := make([]*types.Transaction, 0)
+		appendMap := func(m map[common.Address][]*txpool.LazyTransaction) {
+			for _, list := range m {
+				for _, ltx := range list {
+					if ltx == nil || ltx.Tx == nil {
+						continue
+					}
+					txs = append(txs, ltx.Tx)
+				}
+			}
+		}
+		appendMap(prioPlainTxs)
+		appendMap(prioBlobTxs)
+		appendMap(normalPlainTxs)
+		appendMap(normalBlobTxs)
+
+		// Filter txs for warming and start warm reader cache
+		if len(txs) > 0 && w.chain.WarmInWorkerEnabled() {
+			selectedPrio := selectHeavyAccounts(prioPlainTxs, prioBlobTxs, env.header.GasLimit)
+			selectedNormal := selectHeavyAccounts(normalPlainTxs, normalBlobTxs, env.header.GasLimit)
+
+			// Filter txs slice to only include selected accounts
+			filteredTxs := make([]*types.Transaction, 0)
+			for _, tx := range txs {
+				from, _ := types.Sender(env.signer, tx)
+				if _, ok := selectedPrio[from]; ok {
+					filteredTxs = append(filteredTxs, tx)
+				} else if _, ok := selectedNormal[from]; ok {
+					filteredTxs = append(filteredTxs, tx)
+				}
+			}
+			txs = filteredTxs
 
-		if err := w.commitTransactions(env, plainTxs, blobTxs, interrupt, new(uint256.Int)); err != nil {
-			return err
+			// start warm reader cache
+			tasks, _ := core.NewWarmExecTasks(w.chain, w.chainConfig, env.header, env.state, env.coinbase, txs, vm.Config{})
+			numProcs := max(1, 4*runtime.NumCPU()/5)
```

Comment:
How is this calculation derived?

Reply:
I think this was the same value being used in state prefetcher from upstream.